amostra/objects.py

```diff
@@ -96,12 +96,21 @@ def copy(self):
         d.pop('revision')
         return accessor.new(**d)
 
-    def revert(self, revision):
-        ...
-
     def purge(self):
         self._amostra_client._purge(type(self), self.uuid)
 
+    def revert(self, num):
+        # Find the one you want to revert to
+        for it in self.revisions():
+            if it.revision == num:
+                break
```

Comment:
If no matching revision number is found, this will silently do nothing. You could use the construction

```python
for ...:
    ...
else:
    ...
```

to raise an error if no match is found.